modules/i3s/src/types.ts

```diff
@@ -1,10 +1,71 @@
 import type {GLTFMaterial} from '@loaders.gl/gltf';
-import type {Quaternion, Vector3} from '@math.gl/core';
+import type {Matrix4, Quaternion, Vector3} from '@math.gl/core';
 import type {Mesh} from '@loaders.gl/gltf';
-
-export type Tileset = {[key: string]: any};
+import type {TypedArray} from '@loaders.gl/schema';
+
+export enum DATA_TYPE {
+  UInt8 = 'UInt8',
+  UInt16 = 'UInt16',
+  UInt32 = 'UInt32',
+  UInt64 = 'UInt64',
+  Int16 = 'Int16',
+  Int32 = 'Int32',
+  Int64 = 'Int64',
+  Float32 = 'Float32',
+  Float64 = 'Float64'
+}
+/**
+ * spec - https://github.com/Esri/i3s-spec/blob/master/docs/1.8/3DSceneLayer.cmn.md
+ */
+// TODO Replace "[key: string]: any" with actual defenition
+export type Tileset = {
+  /**
+   * The store object describes the exact physical storage of a layer and enables the client to detect when multiple layers are served from the same store.
+   */
+  store: Store;
+  [key: string]: any;
+};
+// TODO Replace "[key: string]: any" with actual defenition
 export type NodePage = {[key: string]: any};
-export type Tile = {[key: string]: any};
+// TODO Replace "[key: string]: any" with actual defenition
+export type Tile = {
+  content: TileContent;
+  isDracoGeometry: boolean;
+  textureUrl: string;
+  url: string;
+  /**
+   * Resource reference describing a featureData document.
+   */
+  attributeData: Resource[];
+  textureFormat: 'jpeg' | 'png' | 'ktx-etc2' | 'dds' | 'ktx2';
+  textureLoaderOptions: any;
+  materialDefinition: GLTFMaterial;
+  mbs: Mbs;
+};
+// TODO Replace "[key: string]: any" with actual defenition
+export type TileContent = {
+  featureData: DefaultGeometrySchema;
+  attributes: NormalizedAttributes;
+  indices: NormalizedAttribute | null;
+  featureIds: number[] | TypedArray;
+  vertexCount: number;
+  modelMatrix: Matrix4;
+  coordinateSystem: number;
+  byteLength: number;
+  texture: TileContentTexture;
+  [key: string]: any;
+};
+
+export type TileContentTexture =
+  | ArrayBuffer
+  | {
+      compressed: boolean;
+      mipmaps: boolean;
+      width: number;
+      height: number;
+      data: any;
```

Comment:
```suggestion
      data: TextureLevel[];
```
https://github.com/visgl/loaders.gl/blob/d5d1d806740c9ef84496648bd561957cd728c281/modules/textures/src/types.ts#L15